File: clients/node/examples/workflow-connect.js
import { Screen, Pattern } from "../src";

async function main() {
  // Auto-launch workflow.
  const screen = await Screen();
  try {
    const match = await screen.click(Pattern("assets/pattern.png").exact());
    console.log(`clicked match target at (${match.targetX}, ${match.targetY})`);
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: packages/api/cmd/sikuli-go/examples/python/workflow-connect.py
from __future__ import annotations

from bootstrap import ensure_sikuli_go_on_path
from sikuligo import Pattern, Screen

ensure_sikuli_go_on_path()

# Connect-only workflow (requires sikuli-go already running).
screen = Screen.connect()
try:
    match = screen.click(Pattern("assets/pattern.png").exact())
    print(f"clicked match target at ({match.target_x}, {match.target_y})")
finally:
    screen.close()
